matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/DefaultRoomService.kt

```diff
@@ -91,18 +90,25 @@ internal class DefaultRoomService @Inject constructor(
         return roomSummaryDataSource.getRoomSummary(roomIdOrAlias)
     }
 
+    override fun getRoomSummaryLive(roomId: String): LiveData<Optional<RoomSummary>> {
+        return roomSummaryDataSource.getRoomSummaryLive(roomId)
+    }
+
     override fun getRoomSummaries(queryParams: RoomSummaryQueryParams,
                                   sortOrder: RoomSortOrder): List<RoomSummary> {
         return roomSummaryDataSource.getRoomSummaries(queryParams, sortOrder)
     }
 
     override fun refreshJoinedRoomSummaryPreviews(roomId: String?) {
-        val roomSummaries = getRoomSummaries(roomSummaryQueryParams {
-            if (roomId != null) {
-                this.roomId = QueryStringValue.Equals(roomId)
-            }
-            memberships = listOf(Membership.JOIN)
-        })
+        val roomSummaries = if (roomId == null) {
+            getRoomSummaries(roomSummaryQueryParams {
+                memberships = listOf(Membership.JOIN)
+            })
+        } else {
+            listOfNotNull(
+                    getRoomSummary(roomId)?.takeIf { it.membership == Membership.JOIN }
```

Comment:
is there a difference between `listOf(getRoomSummary())` and `getRoomSummaries { this.roomId = QueryStringValue.Equals(roomId) } `?

I'm wondering if all the `roomSummaries` fetching should taking into account `roomIdOrAlias`

Reply:
`RoomSummaryQueryParams.roomId` has been removed, so cannot be used anymore.
`roomIdOrAlias` can be used, but I think for this particular case `roomId` is enough.